test/performance/latency/LatencyTestSubscriber.cpp

```diff
@@ -461,50 +588,80 @@ bool LatencyTestSubscriber::test(
     }
     else
     {
-        latency_type_ = new LatencyType(datasize);
+        latency_type_ = static_cast<LatencyType*>(latency_data_type_->createData());
+        latency_type_->data.resize(datasize, 0);
     }
 
-    std::unique_lock<std::mutex> lock(mutex_);
-    if (command_msg_count_ == 0)
-    {
-        command_msg_cv_.wait(lock);
-    }
-    --command_msg_count_;
+    // Wait for the Publisher READY command
+    unique_lock<mutex> lock(mutex_);
+    command_msg_cv_.wait(
+        lock,
+        [this]()
+        {
+            return command_msg_count_ != 0;
+        });
+    command_msg_count_ = 0;
     lock.unlock();
 
+    // Send to Publisher the BEGIN command
     test_status_ = 0;
     received_ = 0;
     TestCommandType command;
     command.m_command = BEGIN;
-    std::cout << "Testing with data size: " << datasize + 4 << std::endl;
-    command_publisher_->write(&command);
+    if (!command_writer_->write(&command))
+    {
+        logError(LatencyTest, "Subscriber fail to publish the BEGIN command")
+        return false;
+    }
+
+    logInfo(LatencyTest, "Testing with data size: " << datasize + 4);
 
+    // Wait for the STOP or STOP_ERROR commands
     lock.lock();
-    command_msg_cv_.wait(lock, [&]()
-            {
-                return command_msg_count_ > 0;
-            });
-    --command_msg_count_;
+    command_msg_cv_.wait(
+        lock,
+        [this]()
+        {
+            return command_msg_count_ != 0;
+        });
+    command_msg_count_ = 0;
```

Comment:
```suggestion
    --command_msg_count_;
```

Reply:
Done in f2b39f966f52e5b4c9d9b470e07a00c873e79a01